### PR TITLE
fix: disable the checkbox for budget control during payments

### PR DESCRIPTION
**Pull request by Rithvik-padma**
### Describe the changes you have made in this PR

Fixes - #2638 

### Type of change

- `fix`: Bug fix (non-breaking change which fixes an issue)

### Post changes

https://github.com/getAlby/lightning-browser-extension/assets/99183441/6923daa8-eb83-4d69-a749-e405104deea1

### How has this been tested?

Ran all the tests.

### Checklist

- [x] Self-review of changed code
- [x] Manual testing
- [x] Added automated tests where applicable
- [x] Update Docs & Guides
- For UI-related changes
- [x] Darkmode
- [x] Responsive layout

### CODE REVIEW

**Comment by Rithvik-padma** [comment]
@reneaaron I have made the changes for disabling the checkbox and to indicate that it is disabled I have blurred the label and the checkbox.
Please review

File: src/app/screens/ConfirmPayment/index.tsx [hunk 1]
```diff
@@ -173,6 +173,7 @@ function ConfirmPayment() {
                   }}
                   budget={budget}
                   onBudgetChange={(event) => setBudget(event.target.value)}
+                  disabled={true}
```
Comment (reneaaron) [inline src/app/screens/ConfirmPayment/index.tsx:176]: This checkbox should only be disabled while the payment is processing.

File: src/app/components/BudgetControl/index.tsx [hunk 2]
```diff
@@ -27,12 +29,17 @@ function BudgetControl({
...
+        className={`flex items-center ${
+          disabled && "blur-[2px] pointer-events-none select-none"
+        }`}
+      >
         <Checkbox
           id="remember_me"
           name="remember_me"
           checked={remember}
           onChange={onRememberChange}
+          disabled={disabled ? disabled : false}
```
Comment (reneaaron) [inline src/app/components/BudgetControl/index.tsx:42]:
```suggestion
          disabled={disabled}
```

File: src/app/components/BudgetControl/index.tsx [hunk 3]
```diff
@@ -27,12 +29,17 @@ function BudgetControl({
 
   return (
     <div className="mb-4">
-      <div className="flex items-center">
+      <div
+        className={`flex items-center ${
+          disabled && "blur-[2px] pointer-events-none select-none"
```
Comment (reneaaron) [inline src/app/components/BudgetControl/index.tsx:34]: Please remove this blur. 

@stackingsaunter Can you please suggest colors for  disabled form inputs?

**Comment by im-adithya** [comment]
You should disable it _during_ the payment process, also I'd prefer reduced opacity to blur, thoughts @reneaaron?

**Review by reneaaron — DISMISSED** [review]
Thanks very much for the PR! :raised_hand_with_fingers_splayed: 

I've left some first feedback. :ok_hand:

File: src/app/components/BudgetControl/index.tsx (hunk 3 above)
Comment (Rithvik-padma) [reply src/app/components/BudgetControl/index.tsx:34]: Okay will do it

File: .gitignore [hunk 4]
```diff
@@ -151,4 +151,4 @@ dist/
 *.sln
 
 # swc
-.swc/*
```
Comment (rolznz) [inline .gitignore:154]: could you undo the change in this file?

Comment (Rithvik-padma) [reply .gitignore:154]: sure

**Comment by Rithvik-padma** [comment]
@rolznz done the mentioned change, can u review it now.
Thanks!

**Comment by rolznz** [comment]
@Rithvik-padma can you also disable the budget control for keysend payments?

**Comment by rolznz** [comment]
Unfortunately I cannot update this branch. I believe it's because you did not create a branch for this PR, but made a PR from master.

**Comment by Rithvik-padma** [comment]
> @Rithvik-padma can you also disable the budget control for keysend payments? 
> 

Sure will do that!

**Comment by Rithvik-padma** [comment]
> Unfortunately I cannot update this branch. I believe it's because you did not create a branch for this PR, but made a PR from master.

Yes I am using the master branch for it. 
I will make the necessary changes for keysend too and update the branch.
Thanks!

**Comment by Rithvik-padma** [comment]
@rolznz I have disabled the budget control for the keysend payments.
Can you review it.
I did not change the color for the disabled inputs.
@stackingsaunter should I use any color for it or leave it how it is now?
Thanks!

**Review by rolznz — APPROVED** [review]
tACK

I don't believe any styling changes are necessary.

**Comment by rolznz** [comment]
Thanks for your work, @Rithvik-padma !
